### DISCUSSION TITLE
Provide a preview of image transforms with focal points applied

### DISCUSSION THREAD

**Discussion by khalwat** [craftcms/cms#8391] [Ideas]
The FocusPoint plugin does a good job of providing a preview of what the image will look like at various responsive sizes depending on the focal point that is picked:

![ea29baf8-ea0b-11e6-8637-a29d22f919ef](https://cloud.githubusercontent.com/assets/7570798/23821553/ebdfbfb4-0603-11e7-9e3a-0e93faeed883.jpg)

Something akin to this in the image editor would be a boon to content editors, who often don't have a good idea of what the image will look like at various sizes.

**Comment by andris-sevcenko** [comment]
Maybe a preview button. Or maybe allow plugins to add buttons. Worth a discussion at the very least.

**Comment by brandonkelly** [comment]
Yeah the above image is kindof stressful to look at, and seems like a lot of those frames are redundant. But I like the idea in general.

One option is, we give the editor a "Transform Preview" button, which would turn the editor UI into a split view, with the main editor area taking up the left pane, and a preview of the image with a transform applied to it centered in the right pane. The preview pane could have a dropdown at the top where you choose which transform you're looking at - a mix of a few predefined aspect ratios, and if they have any named transforms, we should include those as options too.

@benjamindavid would be great if you could mock that up when you have the time. No rush though, this could wait until after 3.0.
